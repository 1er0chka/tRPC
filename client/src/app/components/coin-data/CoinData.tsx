import React, { FunctionComponent } from "react";
import styles from "./CoinData.module.scss";

interface ICoinDataProps {
  primaryInfo: string;
  secondaryInfo: string;
}

const CoinData: FunctionComponent<ICoinDataProps> = ({
  primaryInfo,
  secondaryInfo,
}) => {
  return (
    <div>
      {secondaryInfo != "$NaN" && secondaryInfo != "&0.00" ? (
        <div className={styles.title}>
          <div className={styles.primaryInfo}>{primaryInfo}</div>
          <div className={styles.secondaryInfo}>{secondaryInfo}</div>
        </div>
      ) : (
        ""
      )}
    </div>
  );
};
export default CoinData;
