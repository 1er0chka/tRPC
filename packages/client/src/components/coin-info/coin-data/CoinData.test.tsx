import {render, screen} from "@testing-library/react";
import CoinData from "./CoinData";

describe('Coin Data Tests', () => {
    test('Snapshot test', () => {
        render(
            <CoinData secondaryInfo={'Secondary info'} primaryInfo={'Primary info'}/>
        )
        const data = screen.queryByTestId('coin-data')
        expect(data).toMatchSnapshot()
    })
    test('Empty test', () => {
        render(
            <CoinData secondaryInfo={"$NaN"} primaryInfo={'Primary info'}/>
        )
        const data = screen.queryByTestId('coin-data')
        expect(data).toMatchSnapshot()
    })
})