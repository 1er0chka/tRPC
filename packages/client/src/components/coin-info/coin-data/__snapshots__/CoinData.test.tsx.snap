// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Data Tests Empty test 1`] = `
<div
  data-testid="coin-data"
/>
`;

exports[`Coin Data Tests Snapshot test 1`] = `
<div
  data-testid="coin-data"
>
  <div
    class="title"
  >
    <div
      class="primaryInfo"
    >
      Primary info
    </div>
    <div
      class="secondaryInfo"
    >
      Secondary info
    </div>
  </div>
</div>
`;
